<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>remaCor: Random effects meta-analysis for correlated test statistics • remaCor</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="remaCor: Random effects meta-analysis for correlated test statistics">
<meta property="og:description" content="remaCor">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">remaCor</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/remaCor.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/simulation_comparison.html">Multivariate transcriptome-wide association study (MulTiWas)</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/GabrielHoffman/remaCor/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="remaCor_files/header-attrs-2.7/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>remaCor: Random effects meta-analysis for correlated test statistics</h1>
                        <h4 class="author"><a href="http://gabrielhoffman.github.io">Gabriel Hoffman</a></h4>
            <address class="author_afil">
      <p>Icahn School of Medicine at Mount Sinai, New York</p>
<br><small class="dont-index">Source: <a href="https://github.com/GabrielHoffman/remaCor/blob/master/vignettes/remaCor.Rmd"><code>vignettes/remaCor.Rmd</code></a></small>
      <div class="hidden name"><code>remaCor.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Standard approaches to meta-analysis assumes that effect sizes are statistically independent. Here we provide methods for fixed and random effects meta-analysis when the correlation between effect sizes are known.</p>
      <p>remaCor v0.0.6<br> April 05, 2021 14:05:50<br></p>
    </div>
    
<!---
cd /Users/gabrielhoffman/workspace/repos
R
library('knitr')
rmarkdown::render('remaCor/vignettes/remaCor.Rmd')
# scp remaCor/vignettes/remaCor.html sklar1:/sc/orga/work/hoffmg01/software/remaCor
--->
<div id="description" class="section level3">
<h3 class="hasAnchor">
<a href="#description" class="anchor"></a>Description</h3>
<div id="fixed-effects-meta-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#fixed-effects-meta-analysis" class="anchor"></a>Fixed effects meta-analysis</h4>
<p><code><a href="../reference/LS.html">LS()</a></code> implements fixed effect meta-analysis for correlated test statistics using method of <span class="citation">Lin and Sullivan (2009)</span>. By default, correlation is set to identity matrix to for independent test statistics.</p>
</div>
<div id="random-effects-meta-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#random-effects-meta-analysis" class="anchor"></a>Random effects meta-analysis</h4>
<p><code><a href="../reference/RE2C.html">RE2C()</a></code> implements random effect meta-analysis for correlated test statistics that jointly tests deviation of the mean from zero as well as effect size heterogenity. This method uses the RE2 method of <span class="citation">Han and Eskin (2011)</span>, or RE2 for correlated test statistics from <span class="citation">Han et al. (2016)</span>. By default, correlation is set to identity matrix to for independent test statistics. (In addition, this function computes the two step RE2C method of <span class="citation">Lee, Eskin, and Han (2017)</span> to further test for heterogenity in effect size after applying a fixed effect test.)</p>
<ul>
<li><p>stat1: statistic testing effect mean</p></li>
<li><p>stat2: statistic testing effect heterogeneity</p></li>
<li><p>RE2Cp: RE2 p-value accounting for correlelation between tests. (This is the p-value appropriate for most questions)</p></li>
<li><p>RE2Cp.twoStep: two step RE2C test after fixed effect test. Only evaluated if twoStep==TRUE. (not typically used)</p></li>
<li><p>QE: test statistic for the test of (residual) heterogeneity</p></li>
<li><p>QEp: p-value for the test of (residual) heterogeneity</p></li>
<li>
<p>Isq: I^2 statistic</p>
<p>QE, QEp and ISq are only evaluted if correlation is diagonal</p>
</li>
</ul>
</div>
</div>
<div id="examples" class="section level3">
<h3 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h3>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/GabrielHoffman/remaCor">remaCor</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://www.metafor-project.org">metafor</a></span><span class="op">)</span>

<span class="co"># Generate effects</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://mvtnorm.R-forge.R-project.org">mvtnorm</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">clusterGeneration</span> <span class="op">)</span>

<span class="va">n</span> <span class="op">=</span> <span class="fl">4</span>
<span class="va">Sigma</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.html">cov2cor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/clusterGeneration/man/genPositiveDefMat.html">genPositiveDefMat</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span><span class="op">$</span><span class="va">Sigma</span><span class="op">)</span>
<span class="va">beta</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/mvtnorm/man/Mvnorm.html">rmvnorm</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">0</span>, <span class="va">n</span><span class="op">)</span>, <span class="va">Sigma</span><span class="op">)</span><span class="op">)</span>
<span class="va">stders</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fl">.1</span>, <span class="va">n</span><span class="op">)</span>

<span class="co"># Standard fixed effects meta-analysis</span>
<span class="co"># of independent effects with metafor pacakge</span>
<span class="fu"><a href="https://wviechtb.github.io/metafor/reference/rma.uni.html">rma</a></span><span class="op">(</span> <span class="va">beta</span>, <span class="va">stders</span>, method<span class="op">=</span><span class="st">"FE"</span><span class="op">)</span>

<span class="co"># Standard random effects meta-analysis</span>
<span class="co"># of independent effects with metafor pacakge</span>
<span class="fu"><a href="https://wviechtb.github.io/metafor/reference/rma.uni.html">rma</a></span><span class="op">(</span> <span class="va">beta</span>, <span class="va">stders</span>, method<span class="op">=</span><span class="st">"REML"</span><span class="op">)</span>

<span class="co"># Run fixed effects meta-analysis, assume identity correlation  </span>
<span class="co"># Use Lin-Sullivan method</span>
<span class="fu"><a href="../reference/LS.html">LS</a></span><span class="op">(</span> <span class="va">beta</span>, <span class="va">stders</span><span class="op">)</span>

<span class="co"># Run fixed effects meta-analysis, accounting for correlation  </span>
<span class="co"># Use Lin-Sullivan method</span>
<span class="fu"><a href="../reference/LS.html">LS</a></span><span class="op">(</span> <span class="va">beta</span>, <span class="va">stders</span>, <span class="va">Sigma</span><span class="op">)</span>

<span class="co"># Run random effects meta-analysis, assume identity correlation  </span>
<span class="fu"><a href="../reference/RE2C.html">RE2C</a></span><span class="op">(</span> <span class="va">beta</span>, <span class="va">stders</span><span class="op">)</span>

<span class="co"># Run random effects meta-analysis, accounting for correlation </span>
<span class="fu"><a href="../reference/RE2C.html">RE2C</a></span><span class="op">(</span> <span class="va">beta</span>, <span class="va">stders</span>, <span class="va">Sigma</span><span class="op">)</span></code></pre></div>
<pre><code>##      stat1    stat2        RE2Cp RE2Cp.twoStep QE QEp Isq
## 1 4.679778 22.57718 4.742258e-07            NA NA  NA  NA</code></pre>
</div>
<div id="session-info" class="section level1">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-apple-darwin19.6.0 (64-bit)
## Running under: macOS Catalina 10.15.7
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libLAPACK.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] grid      stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] mvtnorm_1.1-1           metafor_2.4-0           Matrix_1.3-2            remaCor_0.0.6          
## [5] reshape2_1.4.4          ggplot2_3.3.3           clusterGeneration_1.3.7 MASS_7.3-53.1          
## [9] RUnit_0.4.32           
## 
## loaded via a namespace (and not attached):
##  [1] tidyselect_1.1.0  xfun_0.22         bslib_0.2.4       purrr_0.3.4       lattice_0.20-41  
##  [6] colorspace_2.0-0  vctrs_0.3.6       generics_0.1.0    htmltools_0.5.1.1 yaml_2.2.1       
## [11] utf8_1.2.1        rlang_0.4.10      pkgdown_1.6.1     jquerylib_0.1.3   pillar_1.5.1     
## [16] glue_1.4.2        withr_2.4.1       DBI_1.1.1         lifecycle_1.0.0   plyr_1.8.6       
## [21] stringr_1.4.0     munsell_0.5.0     gtable_0.3.0      ragg_1.1.2        codetools_0.2-18 
## [26] memoise_2.0.0     evaluate_0.14     knitr_1.31        fastmap_1.1.0     fansi_0.4.2      
## [31] Rcpp_1.0.6        scales_1.1.1      cachem_1.0.4      desc_1.3.0        jsonlite_1.7.2   
## [36] systemfonts_1.0.1 fs_1.5.0          textshaping_0.3.3 digest_0.6.27     stringi_1.5.3    
## [41] dplyr_1.0.5       rbibutils_2.0     rprojroot_2.0.2   Rdpack_2.1.1      tools_4.0.3      
## [46] magrittr_2.0.1    sass_0.3.1        tibble_3.1.0      crayon_1.4.1      pkgconfig_2.0.3  
## [51] ellipsis_0.3.1    assertthat_0.2.1  rmarkdown_2.7     R6_2.5.0          nlme_3.1-152     
## [56] compiler_4.0.3</code></pre>
<div id="references" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-han2016general" class="csl-entry">
Han, Buhm, Dat Duong, Jae Hoon Sul, Paul IW de Bakker, Eleazar Eskin, and Soumya Raychaudhuri. 2016. <span>“A General Framework for Meta-Analyzing Dependent Studies with Overlapping Subjects in Association Mapping.”</span> <em>Human Molecular Genetics</em> 25 (9): 1857–66.
</div>
<div id="ref-han2011random" class="csl-entry">
Han, Buhm, and Eleazar Eskin. 2011. <span>“Random-Effects Model Aimed at Discovering Associations in Meta-Analysis of Genome-Wide Association Studies.”</span> <em>The American Journal of Human Genetics</em> 88 (5): 586–98.
</div>
<div id="ref-lee2017increasing" class="csl-entry">
Lee, CH, Eleazar Eskin, and Buhm Han. 2017. <span>“Increasing the Power of Meta-Analysis of Genome-Wide Association Studies to Detect Heterogeneous Effects.”</span> <em>Bioinformatics</em> 33 (14): i379–88.
</div>
<div id="ref-lin2009meta" class="csl-entry">
Lin, Dan-Yu, and Patrick F Sullivan. 2009. <span>“Meta-Analysis of Genome-Wide Association Studies with Overlapping Subjects.”</span> <em>The American Journal of Human Genetics</em> 85 (6): 862–72.
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gabriel E. Hoffman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
